lldb/source/Plugins/Process/Dpu/Dpu.cpp

```diff
@@ -124,7 +123,15 @@ bool Dpu::LoadElf(const FileSpec &elf_file_path) {
   if (status != DPU_OK)
     return false;
 
-  if (!SetPrintfSequenceAddrsFromRuntimeInfo())
+  dpu_program_t *runtime = dpu_get_program(m_dpu);
+  if (!runtime)
+    return false;
+
+  int nr_threads_enabled = runtime->nr_threads_enabled;
+  if (nr_threads_enabled != -1)
```

Comment:
if you have garbage in the environment variable, the strtoll will set this to 0 ?
you might just want to skip 0 either.

Reply:
I don't expect the environment variable to be set by anything else than ``dpu_commands.py``. The variable is indeed not than easy to find, in order to make sure than no one set it by mistake.

Both ``dpu_commands.py`` and the value coming from ``runtime->nr_threads_enabled`` have their default value set to ``-1``.

``runtime->nr_threads_enabled`` is not set by a ``stroll``, but by the value read in the elf.